Assets/MRTK/Core/Providers/BaseController.cs

```diff
@@ -234,50 +234,50 @@ public void AssignControllerMappings(MixedRealityInteractionMapping[] mappings)
         /// <returns>True if a model was successfully loaded or model rendering is disabled. False if a model tried to load but failed.</returns>
         protected virtual bool TryRenderControllerModel(Type controllerType, InputSourceType inputSourceType)
         {
-            if (GetControllerVisualizationProfile() == null ||
-                !GetControllerVisualizationProfile().RenderMotionControllers)
+            MixedRealityControllerVisualizationProfile controllerVisualizationProfile = GetControllerVisualizationProfile();
+            bool controllerVisualizationProfilePresent = controllerVisualizationProfile != null;
+           
+            if (!controllerVisualizationProfilePresent || !controllerVisualizationProfile.RenderMotionControllers)
             {
                 return true;
             }
 
             GameObject controllerModel = null;
+            bool usePlatformModels = !controllerVisualizationProfile.GetUsePlatformModelsOverride(controllerType, ControllerHandedness);
 
-            // If a specific controller template wants to override the global model, assign that instead.
-            if (IsControllerMappingEnabled() &&
-                GetControllerVisualizationProfile() != null &&
-                !(GetControllerVisualizationProfile().GetUseDefaultModelsOverride(controllerType, ControllerHandedness)))
+            // If a specific controller template wants to override the global model, assign it
+            if (!usePlatformModels)
```

Comment:
Are the two `!`s here intended? The original line just checks `!GetUseDefaultModelsOverride`, while this is setting `usePlatformModels` to `!GetUsePlatformModelsOverride` and then checking `!usePlatformModels`

Reply:
good catch, fixed!